### PR TITLE
Isolation level

### PR DESCRIPTION
**Pull request by VictorDenisov**
This PR is for configuring isolation level in batch reading. It's also one of the steps in this story: #216.
There are two variants to configure isolation level either through configuring batch reading, as it's done here, or by configuring connection.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -82,6 +82,19 @@ type ConnConfig struct {
...
+type BatchReadConfig struct {
+	MinBytes       int
+	MaxBytes       int
+	IsolationLevel IsolationLevel
+}
+
+type IsolationLevel int8
+
+const (
+	READ_UNCOMMITTED IsolationLevel = 0
```
Comment (achille-roussel) [inline conn.go]:
```suggestion
	ReadUncommitted IsolationLevel = 0
```

File: conn.go [hunk 2]
```diff
@@ -82,6 +82,19 @@ type ConnConfig struct {
...
+	READ_COMMITTED   IsolationLevel = 1
```
Comment (achille-roussel) [inline conn.go]:
```suggestion
	ReadCommitted   IsolationLevel = 1
```
Let's follow Go naming guidelines ;)

File: conn.go [hunk 3]
```diff
@@ -658,17 +671,26 @@ func (c *Conn) ReadMessage(maxBytes int) (Message, error) {
...
 // the kafka server.
 func (c *Conn) ReadBatch(minBytes, maxBytes int) *Batch {
+	cfg := BatchReadConfig{
+		MinBytes: minBytes,
+		MaxBytes: maxBytes,
+	}
+	return c.ReadBatchWithConfig(cfg)
+}
+
+func (c *Conn) ReadBatchWithConfig(cfg BatchReadConfig) *Batch {
```
Comment (achille-roussel) [inline conn.go]:
```suggestion
func (c *Conn) ReadBatchWith(cfg ReadBatchConfig) *Batch {
```
- We have `NewConnWith` already, so let's match that naming scheme and call the method `ReadBatchWith`?
- To follow these naming schemes, how about also calling the configuration type `ReadBatchConfig`?

File: conn.go [hunk 4]
```diff
@@ -658,17 +671,26 @@ func (c *Conn) ReadMessage(maxBytes int) (Message, error) {
 // gives the minimum and maximum number of bytes that it wants to receive from
 // the kafka server.
 func (c *Conn) ReadBatch(minBytes, maxBytes int) *Batch {
+	cfg := BatchReadConfig{
```
Comment (achille-roussel) [inline conn.go]: This temporary variable doesn't seem useful, how about just inlining the declaration in argument to the method call?

**Comment by VictorDenisov** [comment]
Done

**Comment by achille-roussel** [comment]
> Could we add documentation for the new types and method that this PR adds?

**Comment by VictorDenisov** [comment]
Ah, of course. I will.

**Comment by VictorDenisov** [comment]
@achille-roussel I added some docs without documenting most obvious(in my opinion) things. Feel free to let me know if you want more docs.

File: write.go [hunk 5]
```diff
@@ -196,7 +196,7 @@ func writeFetchRequestV5(w *bufio.Writer, correlationID int32, clientID, topic s
 	writeInt32(w, milliseconds(maxWait))
 	writeInt32(w, int32(minBytes))
 	writeInt32(w, int32(maxBytes))
-	writeInt8(w, int8(0)) // isolation level 0 - read uncommitted
+	writeInt8(w, isolationLevel) // isolation level 0 - read uncommitted
```
Comment (achille-roussel) [inline write.go]:
```suggestion
	writeInt8(w, isolationLevel)
```

**Pull request merged**